### ISSUE TITLE
Review updates to Example Page for Tabs With Manual Activation 

### ISSUE THREAD

**Issue by mcking65** [w3c/aria-practices#279]
This issue tracks review of final editorial, template, and functional updates to the 
[Example of Tabs With Manual Activation](http://w3c.github.io/aria-practices/examples/tabs/tabs-2/tabs.html).

#### Requested Reviews

- [x] Review by @MichielBijl  
- [x] Review by @annabbott 
- [x] Review by @accdc 
- [x] Review by @shirsha 

#### Summary of changes

1. Move examples.js script element to head and fix src attribute path so HTML source will display.
2. Make H1 content match page title.
3. Editorial revisions to introduction.
4. Correct href of link to tabs design pattern in introduction.
5. Added link to section that provides guidance on selection follows focus.
6. Per example template, Removed region around example and replaced with screen reader sepearator elements.
7. Added `aria-label="Entertainment"` to the tablist.
8. Added `tabindex="0"` to the tabpanel elements.
9. Added information about the delete function to the accessibility features section.
10. Editorial revisions to the keyboard table.
11. In roles, states and properties table:
    * added scope attribute and TH elements per template.
    * Added row for aria-label on tablist
    * Added rows for aria-selected true and false on tab
    * Added row for aria-labelledby on tabpanel
    * Added row for tabindex on tabpanel.
12. Per template, Added screen reader separators around HTML source display and removed role region from the source code display div.
13. Corrected href attribute of link to tabs design pattern in footer nav.

**Comment by MichielBijl** [comment]
> Remove delete from keyboard table since it is not supported.

Delete is supported, but only on the last tab (same as in the other example). We can mention that in the text if that’d make it clearer.

**Comment by mcking65** [comment]
I restored information about the delete function and added it to the accessibility features section. I am also editing the head comment in this issue to reflect this.

I noticed, however, that focus is not correctly set after delete so raised issue #286.

**Comment by mcking65** [comment]
@michielBijl, thank you very much for the speedy resolution of issue #286!!!!!

**Comment by MichielBijl** [comment]
No problem @mcking65 :).

**Comment by MichielBijl** [comment]
I noticed that [design we used wasn’t usable in Windows High Contrast mode](https://github.com/w3c/aria-practices/issues/289). So I’ve made a new design that does. This doesn’t impact the functionality or HTML code.

**Comment by annabbott** [comment]
When navigating the page using only the keyboard in FF, Tab keystroke activates the "Nils Frahm" tab. I expected the next Tab keystroke to exit the widget and put focus on the first link that follows the tab panel. Instead, the next Tab keystroke puts focus on the tabpanel boundary. Same behavior occurs when the "Agnes Obel" tab is activated and I press Tab key.

**Comment by mcking65** [comment]
@annabbott, focusing the tabpanel is intentional as described in the keyboard section.

This is an optional behavior. Do we need guidance to suggest when it is advisable?

**Comment by annabbott** [comment]
I don't see that moving focus to the tabpanel is "optional" nor do I see any guidance regarding when it is advisable. Both are needed.

**Comment by mcking65** [comment]
@Annabbott, your concerns will be resolved at a later time with issues #322 and #323.

Still looking for feedback from @accdc and @shirsha.

**Comment by shirsha** [comment]
Looks good. 
Working as per design pattern.

**Comment by accdc** [comment]
Hi, I've gone through this, and it seems to me that for a manual activation tab control there needs to be a distinction between an expanded tab versus a selected one, which this paradigm does not address.

E.G There are some instances where multiple tabs may be selected in order to perform a batch operation on all at once such as deleting 3 out of 5 tabs, which is a totally separate action from expanding one specific tab for viewing. This may be accomplished by holding down the control key while moving using the arrow keys and pressing Ctrl+Space to toggle aria-selected on the focused tab in order to toggle selection, or pressing Space or Enter by itself will expand the currently focused tab instead.

In a case such as this, then aria-selected cannot be used to indicate which tab is open, but aria-expanded must instead be used to convey this, and there needs to be an example of this within this archive if not within this specific example.

Here is an example that shows this particular difference where aria-selected is separate from aria-expanded:

http://whatsock.com/tsg/Coding%20Arena/ARIA%20and%20Non-ARIA%20Tabs/ARIA%20Tabs%20(Internal%20Content)/demo.htm

**Comment by accdc** [comment]
That's fine.

From your comments there seems to be a bias to make sure that people not use aria-expanded on tabs. Why is this when aria-expanded is a supported attribute on role=tab and works across all platforms?

**Comment by mcking65** [comment]
@accdc commented:

> From your comments there seems to be a bias to make sure that people not use aria-expanded on tabs. Why is this when aria-expanded is a supported attribute on role=tab and works across all platforms?

That's a really good question that we have not really discussed much.

To be sure aria-expanded is allowed, and it was the technique we had in mind during the drafting of ARIA 1.0 when thinking about using tab elements to represent accordion headers -- a technique we have since abandoned.

I am leaning toward aria-current instead because:

1. When we wrote the aria-current spec, we specifically suggested that it applies to this scenario.
2. aria-current can be used to communicate this semantic in patterns where aria-expanded has other meaning (e.g., in trees). So by using aria-current, we can create more semantic consistency across patterns.
3. It seems the semantics more closely align with what needs to be communicated.
    * aria-current: Indicates the element that represents the current item within a container or set of related elements.
    * aria-expanded: Indicates whether the element, or another grouping element it controls, is currently expanded or collapsed.

In a set of tabs, where one panel can be displayed, we are communicating which one in the set is displayed. A navigation tree, menu, listbox, grid, or even static list of links could do the same. 

Tree branches, submenus, disclosure panels, and accordion panels (in most cases), can all "appear" and "disappear" independently -- now you see it -- now you don't. I understand the aria-current semantic to be saying that something is always current and this is the one that is now current (or active).

This is all sort of subtle and squishy, but I think, hope, that it is clear enough that assistive technologies could one day exploit it when helping users understand a user interface.

Given the variety of interpretations of ARIA that have developed over the last 5 years, assistive technologies have understandably avoided drawing too many conclusions from ARIA. On some pages an ARIA attribute means one thing and on another page the same attribute means something else and on other pages the attribute snuck in accidentally and has no meaning at all. So, the one day I speak of may be a ways off. It's our job here to build community consensus around what that day is like and to do what we can to accelerate its arrival.
